modules/fingerprint.py

```diff
@@ -48,47 +51,71 @@ def scan(self):
+        sorted_boot_fingerprints = sorted(self.boot_fingerprints.items(), key=operator.itemgetter(0))
+        for filename, filemd5 in sorted_boot_fingerprints:
             fingerprints += filemd5 + "\t" + filename + "\n"
         return fingerprints
 
-    def getFingerPrints(self):
-        return self.fingerprints
+    def getRootFingerPrints(self):
+        return self.root_fingerprints
 
-    def validateFingerPrints(self):
-        toReturn = True
-
-        if len(self.fingerprints) == 0:
-            self.scan()
-
-        # Compute default file for fingerprints
-        defaultFingerPrintFile = getConfigurationItem(self.conf, "FingerPrintScanner", "defaultFingerPrintFile")
-        root_mount = getConfigurationItem(self.conf, 'General', 'host_bind_mount')
-        if not root_mount:
-            root_mount = '/'
-        defaultFingerPrintFile = os.path.join(root_mount, defaultFingerPrintFile)
+    def getBootFingerPrints(self):
+        return self.boot_fingerprints
 
-        if defaultFingerPrintFile and os.path.isfile(defaultFingerPrintFile):
-            # Host OS fingerprint present on filesystem
-            with open(defaultFingerPrintFile) as infile:
+    def do_validateFingerPrints(self, fingerPrintFile, fingerprints):
+        ''' Compares the computed fingerprints in <fingerprints> with the one in the <fingerPrintFile> file '''
+        toReturn = True
+        try:
+            with open(fingerPrintFile, 'r') as infile:
                 for line in infile:
                     default_filename = line.split()[1]
                     default_filemd5 = line.split()[0]
-                    for filename,filemd5 in self.fingerprints.items():
+                    for filename,filemd5 in fingerprints.items():
                         if filename == default_filename and filemd5 != default_filemd5:
                             log.warn("Fingerprint failed for: " + filename)
                             toReturn = False
+        except Exception as e:
+            print(e)
+            return False
+        return toReturn
+
+    def validateFingerPrints(self):
+        ''' Verifies if computed fingerprints match the ones in the fingerprints file'''
+        toReturn = True
+
+        # If we didn't computed fingerprints do it now
+        if len(self.root_fingerprints) == 0 or len(self.boot_fingerprints) == 0:
+            self.scan()
+
+        # Compute fingerprints file for rootfs partition
+        root_FingerPrintFile = getConfigurationItem(self.conf, "FingerPrintScanner", "root_defaultFingerPrintFile")
+        root_FingerPrintFile = os.path.join(self.root, root_FingerPrintFile)
+
+        # Validate root partition fingerprints
+        if os.path.isfile(root_FingerPrintFile):
+            toReturn = self.do_validateFingerPrints(root_FingerPrintFile, self.root_fingerprints)
         else:
-            # Host OS fingerprint not present on filesystem
+            # Rootfs fingerprints file not present on filesystem
```

Comment:
If there is no such file, we try to match an existing one from the set to ship with the tool. Check the fllow up code and https://github.com/resin-os/resinhup/tree/master/modules/fingerprint/known-images